### ISSUE TITLE
[Accordion] allow state to expand or collapse tabs without nullifying allowToggle and blocking user interaction

### ISSUE THREAD

**Issue by amfriedman** [chakra-ui/chakra-ui#5176]
### Description

TL;DR
**index** nullifies **allowToggle**, and **defaultIndex** cannot be set by state. This is bad.

Full:
I think there is a fundamental flaw in the architecture of the Chakra UI Accordion component.

I am setting **defaultIndex** to a value that gets set after a useEffect() on a routing component according to items in a permalink URL that changes as the user navigates my app. 

Because I am setting **defaultIndex** in this way, Accordion ignores it and nothing happens. No panels open. This is because the behavior is to honor **defaultValue** only when Accordion is first rendered, and setting this prop via state later is simply ignored.

**index** on the other hand, does change with a state-set value as expected. However, the problem is, once **index** is set, all accordion panels are "locked". Even if **allowToggle** = true, panels cannot be freely expanded or collapsed by the user (which seems to contradict the promised behavior in the documentation).

### Problem Statement/Justification

It seems like a common use case to want the application to open or close accordion panels programmatically, but **not** block the user from also opening and closing panels.

### Proposed Solution or API

A few potential solutions (in order of most to least elegant IMO):

1. Change the behavior of allowToggle to actually do what it is promised to do in the docs: that is, allow the user to toggle panels even if they are set to be open/closed programmatically. 
2. Change the behavior of **defaultIndex** so it renders changes to panels when its value is changed via state.
2. Add another prop called **indexSoft** which specifically does not block user interactivity going forward.

**Comment by segunadebayo** [comment]
Hi @amfriedman,

> I am setting defaultIndex to a value that gets set after a useEffect() on a routing component according to items in a permalink URL that changes as the user navigates my app.

This isn't a design flaw in Chakra UI, perhaps it's a slight misunderstanding on your part. `defaultIndex` and `index` works just like `defaultValue` and `value` in native elements like `input`, `select`, etc.

`defaultIndex`  can only be see initially and can't be updated afterward. To control the accordion programmatically, I'd recommend using the `index` and `onChange` prop as described in the docs.

Please provide a CodeSandbox reproduction if you think it's still an issue and I'll re-open it.

**Comment by Mounajidev** [comment]
Hello !, 
 would you mind please to explain how could achieve, that when an item inside the Accordion Panel is selected, it collapse automatically,  so that the user dont need to go and click again on the accordion button to collapse it ?.

**Comment by sleekLancelot** [comment]
> Hello !, would you mind please to explain how could achieve, that when an item inside the Accordion Panel is selected, it collapse automatically, so that the user dont need to go and click again on the accordion button to collapse it ?.

Hi does anyone know the answer to this? cant find it anywhere in the docs, I want to close the accordion using a button inside of it. What function do i call?

**Comment by sleekLancelot** [comment]
> > Hello !, would you mind please to explain how could achieve, that when an item inside the Accordion Panel is selected, it collapse automatically, so that the user dont need to go and click again on the accordion button to collapse it ?.
> 
> Hi does anyone know the answer to this? cant find it anywhere in the docs, I want to close the accordion using a button inside of it. What function do i call?

Incase anyone is still looking, i solved mine by wrapping the button with `<AccordionButton>` tag, like so:
```
<h6>
	<AccordionButton>
		<Button>Cancel</Button>
	</AccordionButton>
</h6>
```

Note: this button is inside the accordion it intends to close.

**Comment by TravisHi** [comment]
> 

Thanks, this is exactly what I was after!

**Comment by Razakhan45** [comment]
> > > Hello !, would you mind please to explain how could achieve, that when an item inside the Accordion Panel is selected, it collapse automatically, so that the user dont need to go and click again on the accordion button to collapse it ?.
> > 
> > 
> > Hi does anyone know the answer to this? cant find it anywhere in the docs, I want to close the accordion using a button inside of it. What function do i call?
> 
> Incase anyone is still looking, i solved mine by wrapping the button with `<AccordionButton>` tag, like so:
> 
> ```
> <h6>
> 	<AccordionButton>
> 		<Button>Cancel</Button>
> 	</AccordionButton>
> </h6>
> ```
> 
> Note: this button is inside the accordion it intends to close.
@ sleekLancelot
the problem with doin this is that default index Property no longer work properly .
Defining another Accordion Button inside the panel cause this error . Because according Panels open and close based on their Buttons .
if two button in one accordion item then index / DefaultIndex work when given 1 value where it should be 0 because of two button.
